source/parsed_grid_generator.cc

```diff
@@ -232,62 +232,146 @@ ParsedGridGenerator<dim, spacedim>::serial()
   return tria;
 }
 
+namespace
+{
+  template<int dim, int spacedim>
+  bool
+  default_create_grid( Triangulation<dim,spacedim> &tria,
+                       std::string grid_name,
+                       Point<spacedim> point_option_one,
+                       Point<spacedim> point_option_two,
+                       double double_option_two,
+                       double double_option_one,
+                       bool bool_option_one,
+                       unsigned int un_int_option_one,
+                       std::vector<unsigned int> un_int_vec_option_one,
+                       std::string input_grid_file_name)
+  {
+    if (grid_name == "rectangle")
+      {
+        GridGenerator::subdivided_hyper_rectangle (tria,
+                                                   un_int_vec_option_one,
+                                                   point_option_two,
+                                                   point_option_one,
+                                                   bool_option_one);
+        return true;
+      }
+    else if (grid_name == "file")
+      {
+        GridIn<dim, spacedim> gi;
+        gi.attach_triangulation(tria);
+
+        std::ifstream in(input_grid_file_name.c_str());
+        AssertThrow(in, ExcIO());
+
+        std::string ext = extension(input_grid_file_name);
+        if (ext == "vtk")
+          gi.read_vtk(in);
+        else if (ext == "msh")
+          gi.read_msh(in);
+        else if (ext == "ucd" || ext == "inp")
+          gi.read_ucd(in);
+        else if (ext == "unv")
+          gi.read_unv(in);
+        else
+          Assert(false, ExcNotImplemented());
+        return true;
+      }
+    return false;
+  }
+
+  template<int dim, int spacedim>
+  bool
+  create_grid(Triangulation<dim,dim+1> &tria,
+              std::string grid_name,
+              Point<spacedim> point_option_one,
+              Point<spacedim> point_option_two,
+              double double_option_two,
+              double double_option_one,
+              bool bool_option_one,
+              unsigned int un_int_option_one,
+              std::vector<unsigned int> un_int_vec_option_one)
+  {
+    return false;
```

Comment:
Can you also implement the hyper_sphere? Nicola C added it to the library... since you're at it, it should not be difficult...